ArcGISRuntimeSDKQt_CppSamples/LocalServer/LocalServerFeatureLayer/LocalServerFeatureLayer.cpp

```diff
@@ -60,37 +61,73 @@ void LocalServerFeatureLayer::componentComplete()
+  // Check for ArcGIS Pro map package files
+  QString fileName = "PointsofInterest.mpkx";
+  QString dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpkx/" + fileName;
+
+  // Check to see if map package exists
+  if (!QFileInfo::exists(dataPath) || !QFileInfo(dataPath).isFile())
+  {
+    qDebug() << "ArcGIS Pro .mpkx file not found at" << dataPath << "\nChecking for .mpk file";
+    fileName = "PointsofInterest.mpk";
+    dataPath = QDir::homePath() + "/ArcGIS/Runtime/Data/mpk/" + fileName;
+    if (!QFileInfo::exists(dataPath) || !QFileInfo(dataPath).isFile())
+      qDebug() << "File:" << dataPath << "not found";
+    else
+      qDebug() << "Using .mpk file from" << dataPath;
+  }
 
   // create a feature service
-  m_localFeatureService = new LocalFeatureService(dataPath + "/mpk/PointsofInterest.mpk", this);
+  m_localFeatureService = new LocalFeatureService(dataPath, this);
 
   if (LocalServer::instance()->isInstallValid())
   {
     connectSignals();
     if (LocalServer::status() == LocalServerStatus::Started)
-      m_localFeatureService->start();
-    else
+    {
+      qDebug() << "Local server was already running";
+
+      // The local server can already be running if it was started in a different application.
+      // We start up the local feature service here because there will be no status change signal to otherwise trigger it.
+
+      if (m_localFeatureService->status() != LocalServerStatus::Started || m_localFeatureService->status() != LocalServerStatus::Starting)
+      {
+        m_localFeatureService->start();
+      }
+    }
+    else if (LocalServer::status() != LocalServerStatus::Started || LocalServerStatus() != LocalServerStatus::Starting)
+    {
       LocalServer::start();
+    }
   }
+  else
+    qDebug() << "Local server install invalid";
 }
 
 void LocalServerFeatureLayer::connectSignals()
 {
-  // local server status
+  // Local server status
   connect(LocalServer::instance(), &LocalServer::statusChanged, this, [this]()
   {
+    // If the local server was not previously running, our local feature service will start here if the local server has successfully started.
     if (LocalServer::status() == LocalServerStatus::Started)
     {
-      qDebug() << "Local server started";
+      if (m_localFeatureService->status() != LocalServerStatus::Started || m_localFeatureService->status() != LocalServerStatus::Starting)
+        m_localFeatureService->start();
     }
   });
 
   // local feature service status
   connect(m_localFeatureService, &LocalFeatureService::statusChanged, this, [this]()
   {
-    if (m_localFeatureService->status() == LocalServerStatus::Started)
+    if (m_localFeatureService->status() == LocalServerStatus::Starting)
+    {
+      qDebug() << "Local feature service starting up";
+    }
+    else if (m_localFeatureService->status() == LocalServerStatus::Started)
     {
+      qDebug() << "Successfully hosting local feature service at:" << m_localFeatureService->url().toString();
```

Comment:
I'll keep that in mind when I have to insert multiple arguments into a string. The "Realistic lighting and shadows" sample has a good instance of where this can work well.